test/channel_test.dart

```diff
@@ -53,4 +144,62 @@ void main() {
       expect(callbackCalled, 3);
     });
   });
+
+  group('off', () {
+    setUp(() {
+      socket = RealtimeClient('/socket');
+
+      channel = socket.channel('topic', chanParams: {'one': 'two'});
+    });
+
+    test('removes all callbacks for event', () {
+      var callBackEventCalled1 = 0;
+      var callbackEventCalled2 = 0;
+      var callbackOtherCalled = 0;
+
+      channel.on(
+          'event', (dynamic payload, {String? ref}) => callBackEventCalled1++);
+      channel.on(
+          'event', (dynamic payload, {String? ref}) => callbackEventCalled2++);
+      channel.on(
+          'other', (dynamic payload, {String? ref}) => callbackOtherCalled++);
+
+      channel.off('event');
+
+      channel.trigger('event', payload: {}, ref: defaultRef);
+      channel.trigger('other', payload: {}, ref: defaultRef);
+
+      expect(callBackEventCalled1, 0);
+      expect(callbackEventCalled2, 0);
+      expect(callbackOtherCalled, 1);
+    });
+  });
+
+  group('unsubscribe', () {
+    setUp(() {
+      socket = RealtimeClient('/socket');
+
```

Comment:
These `unsubscribe` tests fail, and I am having a hard time figuring out why.

Reply:
cos channel is only [removed](https://github.com/supabase/realtime-dart/blob/1583d3a6d39cb983b7ef42096888448ed67403bc/lib/src/realtime_subscription.dart#L45) when there is a response from realtime server on [ChannelEvents.leave](https://github.com/supabase/realtime-dart/blob/1583d3a6d39cb983b7ef42096888448ed67403bc/lib/src/realtime_subscription.dart#L141) Push event.

Reply:
you may use `mocktail` to provide mock for realtime server like [socket_test](https://github.com/supabase/realtime-dart/blob/main/test/socket_test.dart)